runtime/compiler/z/codegen/J9TreeEvaluator.hpp

```diff
@@ -458,13 +458,14 @@ class OMR_EXTENSIBLE TreeEvaluator: public J9::TreeEvaluator
    /*
     * Generate instructions for static/instance field access report.
     */ 
-   static void generateTestAndReportFieldWatchInstructions(TR::CodeGenerator *cg, TR::Node *node, TR::Snippet *dataSnippet, bool isWrite, TR::Register *sideEffectRegister, TR::Register *valueReg);
+   static void generateTestAndReportFieldWatchInstructions(TR::CodeGenerator *cg, TR::Node *node, TR::Snippet *dataSnippet, bool isWrite, TR::Register *sideEffectRegister, TR::Register *valueReg, TR::Register *dataSnippetRegister);
```

Comment:
This new argument (`dataSnippetRegister`) that we are introducing will only be used by the Power code generator. Doing so provides a simple solution and will allow all platforms to share as much code as possible.

Considering the above, I think it'll be good if we update the doxygen documentation in a platform agnostic way so that anyone looking at this code in the future will not question why X and Z implementations accept an argument that they do not use. Example, something along the lines of (perhaps it can be worded better):
```
\params
...
dataSnippetRegister - If needed, the dataSnippet's address can be populated in this register.
```

Reply:
Additionally, can we document this API change in https://github.com/eclipse/openj9/issues/2668

Reply:
Right. I'll update the doxygen documentation. As well as add it to #2668

Reply:
@AlenBadel Why was this resolved? I don't see the update on #2668. Can we leave this as unresolved until the update is there for tracking purposes?